pkg/datasources/datasource_system_parameter.go

```diff
@@ -0,0 +1,89 @@
+package datasources
+
+import (
+	"context"
+
+	"github.com/MaterializeInc/terraform-provider-materialize/pkg/utils"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func SystemParameter() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: systemParameterRead,
+		Schema: map[string]*schema.Schema{
+			"parameters": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The name of the system parameter.",
+						},
+						"setting": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The value of the system parameter.",
+						},
+						"description": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The description of the system parameter.",
+						},
+					},
+				},
+			},
+			"region": RegionSchema(),
+		},
+	}
+}
+
+func systemParameterRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var diags diag.Diagnostics
+
+	metaDb, region, err := utils.GetDBClientFromMeta(meta, d)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	conn := metaDb
+
+	rows, err := conn.Query("SHOW ALL;")
```

Comment:
Currently the system parameter data source gets all available parameters using `show all;`. We could change this and come up with a query to only get specific parameters if we think that this might be too much.

Reply:
Would that be a breaking change, or could it be introduced in a backwards-compatible way? Are we concerned about performance w/a `SHOW ALL;`?

Reply:
Yep, it should be ok as long as we keep the same schema and only change the underlying SQL statement. 

My main concern with `show all` is that it returns a lot of information and most of it might not be that helpful for the user. Though I might be looking at it wrong, and it might actually be better to have that information available in case that a user ever needs it.

Reply:
Would it complicate debugging/logging? e.g., if we need to go through verbose logs we'd need to filter out the same wall of text for each invocation?

Reply:
Nope that will not be the case, the output of the query is not added to the debug output.